store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/utils/TableUtils.java

```diff
@@ -216,7 +221,63 @@ public static Connector getConnector(final String instanceName, final String zoo
         try {
             return instance.getConnector(userName, new PasswordToken(password));
         } catch (final AccumuloException | AccumuloSecurityException e) {
-            throw new StoreException("Failed to create accumulo connection", e);
+            throw new StoreException("Failed to create accumulo connection (using password-based authentication)", e);
+        }
+    }
+
+    /**
+     * Creates a connection (using Kerberos authentication) to
+     * an accumulo instance using the provided parameters
+     *
+     * @param instanceName the instance name
+     * @param zookeepers   the zoo keepers
+     * @param principal    the Kerberos principal
+     * @param keytabPath   the path to the Kerberos keytab
+     * @return A connection to an accumulo instance
+     * @throws StoreException failure to create an accumulo connection
+     */
+    public static Connector getConnectorKerberos(final String instanceName, final String zookeepers,
+                                                 final String principal, final String keytabPath) throws StoreException {
+        final Instance instance = new ZooKeeperInstance(instanceName, zookeepers);
+        try {
+            // Configure Hadoop UGI to use Kerberos (if it's not already)
+            if (!UserGroupInformation.isSecurityEnabled()) {
+                Configuration conf = new Configuration();
+                conf.set("hadoop.security.authentication", "kerberos");
```

Comment:
Use a string constant.

Reply:
I would, except that would add an extra two lines, 200 odd lines above here and wouldn't provide any benefit because these values are not going to be used anywhere else in the class or in Gaffer.